### PR TITLE
London Class 9 - Farzaneh Haghani - JS2 - Week 4

### PR DESCRIPTION
**Pull request by farzaneh-haghani**
# Your Details

- Your Name: Farzaneh Haghani
- Your City: London
- Your Slack Name: Farzaneh Haghani

# Homework Details

- Module: JavaScript 2
- Week: 4

# Notes

- What did you find easy? setTimeout && setInterval

- What did you find hard? nothing

- What do you still not understand? nothing

- Any other notes? no

### CODE REVIEW

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 1]
```diff
@@ -1,4 +1,49 @@
...
+    if (second === -1 && minute > 0) {
+      minute--;
+      second = 59;
+    }
+
+    if (second < 10)
+      second = second.toLocaleString('en-US', {
+        minimumIntegerDigits: 2,
+        useGrouping: false
+      })
```
Comment (JDysiewicz) [inline 1-mandatory/1-alarmclock/alarmclock.js:21]: nice use of existing functionality to leftpad the string

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 2]
```diff
@@ -1,4 +1,49 @@
...
+
+    if (second >= 0) {
+      timeRemaining.innerText = `Time Remaining: ${minute}:${second}`;
+      second--;
+    }
+
+    if (second === -1 && minute === "00") {
```
Comment (JDysiewicz) [inline 1-mandatory/1-alarmclock/alarmclock.js:34]: This works, though generally conditionals should be checking logically relevant conditions - what you care about here is when the overall time reaches `0`, so checking that a string equals some string and that seconds is negative makes this a little hard to understand. 

I'd recommend converting the entered time into seconds and keeping track of that as your method of tracking time. This means you can use existing JS libraries to format things for you (like the `Date` object`:
```
const formattedTime  = new Date(seconds * 1000).toISOString().slice(11, 19);
// outputs string like `"hh:mm:ss"` e.g. `"00:09:12"` for 9min12sec
```

you can then make this (seemingly random line of code) make sense by giving it a context-appropriate function name/comment:
```
// formats time in "hh:mm:ss" format
const formatTime = (seconds) => {
   return new Date(seconds * 1000).toISOString().slice(11, 19);
}
```

File: 1-mandatory/2-quotegenerator/quotes.js [hunk 3]
```diff
@@ -490,3 +490,61 @@ const quotes = [
...
+check.setAttribute("type", "checkbox");
+
+document.body.appendChild(div);
+div.appendChild(labelToggle);
+labelToggle.appendChild(spanStatus);
+labelToggle.appendChild(check);
+labelToggle.appendChild(spanSlider);
+div.appendChild(pQuote);
+div.appendChild(pAuthor);
+div.appendChild(button);
```
Comment (JDysiewicz) [inline 1-mandatory/2-quotegenerator/quotes.js:512]: see what I mentioned before how you're already grouping these into groups of functionality, you can split this into a seperate function with a nice name which explains what this grouping does

File: 1-mandatory/3-slideshow/slideshow.js [hunk 4]
```diff
@@ -1 +1,68 @@
...
+    }, delayTime * 1000)
+})
+
+function backShow() {
+    i = (i + images.length - 1) % images.length;
+    img.src = `./assets/${images[i]}`;
+}
+
+function forwardShow() {
+    i = (i + 1) % images.length;
```
Comment (JDysiewicz) [inline 1-mandatory/3-slideshow/slideshow.js:66]: nice use of `%`

**Review by JDysiewicz — COMMENTED** [review]
Not too much to add, all looks really good. If you start grouping statements into groups, that's an indicator you could probably turn those statements into a separate function

File: 1-mandatory/1-alarmclock/alarmclock.js [hunk 5]
```diff
@@ -32,12 +33,19 @@ function setAlarm() {
         second--;
       }
 
-      if (second === -1 && minute === "00")
+      if (second === -1 && minute === "00") {
+        let i = 0;
+        const stopFlashing = setInterval(() => {
+          document.body.style.backgroundColor = flash[i++ % flash.length];
```
Comment (ebrahimbeiati) [inline 1-mandatory/1-alarmclock/alarmclock.js]: good job

**Pull request closed without merging**